force-app/main/default/classes/MessagingLWCController.cls

```diff
@@ -0,0 +1,70 @@
+public with sharing class MessagingLWCController {
+    
+    @AuraEnabled(cacheable = true)
+    public static MessagingWrapper getMessages(String contactId, String offSetValFromUI){
+        	MessagingWrapper msgWrapper = new MessagingWrapper();
+        	Integer totalMessagesCount = [SELECT Count() FROM Message__c WHERE Contact__c = :contactId];
+        	Integer offSetVal=0;
+        	
+        	String strQuery = 'Select Id,Is_Outbound__c, Message_Content__c, CreatedBy.name, CreatedDate,Is_Delivered__c,Contact__r.Name FROM Message__c WHERE Contact__c = \'' + contactId + '\'';
+        	
+            if(totalMessagesCount > Integer.ValueOf(Label.MessagePerPage)){
+                if(String.IsNotBlank(offSetValFromUI)){
+                       if(Integer.valueOf(offSetValFromUI) >= 0){
+                           offSetVal = Integer.Valueof(offSetValFromUI);
+                       }else{
+                           offSetVal=0;
+                       }
+                }else{
+                    offSetVal = totalMessagesCount - Integer.ValueOf(Label.MessagePerPage);
+                    strQuery += ' ORDER BY CreatedDate LIMIT '+ Label.MessagePerPage;
+                }
+            }
```

Comment:
Deeply nested if..then statements are hard to read
https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_apex_design.html#avoiddeeplynestedifstmts